### ISSUE TITLE
PyPI release

### ISSUE THREAD

**Issue by paugier** [pyFFTW/pyFFTW#385]
A new PyPI release would be very useful, in particular to add wheels for Python 3.12. However, we might want to wait to do few more things, in particular:

- [x] Support for Numpy 2.0 #384 
- [x] Build wheels for macosx arm64 #382
- [x] Fix documentation build at RTD #387 

And I may forget other important things?

Nevertheless, if we see that these tasks are too difficult and would take too long, I think it make sense to release a new version soon. I would be interested to have the points of view of pyFFTW maintainers/developers on these questions.

**Comment by paugier** [comment]
It seems to me that next version should be 0.13.2, but I may have missed a breaking change... So 0.13.2 or 0.14.0?

**Comment by hgomersall** [comment]
0.14 I think since it breaks backwards compatibility potentially (versions dropped)

**Comment by hgomersall** [comment]
I think we're good for a release? I _think_ this just needs a tag to be pushed - with your  more recent study of the build system, does that sound right?

**Comment by paugier** [comment]
I just realized that there might be an issue here

https://github.com/pyFFTW/pyFFTW/blob/a6a70f3ee52ffb6428317a37954848f08a6f6e29/.github/workflows/wheel_tests_and_release.yml#L272-L275

I guess this might fail because we should not call setup.py directly but it should be used with tools like `pip` or `build` (which use isolated build). I'm going to submit a last PR.

**Comment by paugier** [comment]
After #389 is merged, I think it could indeed be time to push the tag `v0.14.0`! 🤞

**Comment by gabrielfougeron** [comment]
Python 3.12 wheels? Awesome! :-)
Is there something blocking the inclusion of #375 in this release?

**Comment by paugier** [comment]
> Is there something blocking the inclusion of https://github.com/pyFFTW/pyFFTW/pull/375 in this release?

Could you please rebase this PR on master to see what it gives?

**Comment by gabrielfougeron** [comment]
OK, I think I just did. I'm not very familiar with rebase though, so please do check I did not make an obvious mistake.

**Comment by paugier** [comment]
I checked and indeed your rebase is correct :-) Some jobs on Windows fail but this should be easy to fix so I'm confident that https://github.com/pyFFTW/pyFFTW/pull/375 could be included in `0.14.0`.

**Comment by paugier** [comment]
@hgomersall Let's go for `v0.14.0` !?

**Comment by hgomersall** [comment]
Yes. Someone drafted a careful changelog and credits last time to accompany the release. That is probably desirable...

**Comment by hgomersall** [comment]
@grlee77 help! How do I make a pretty release!

**Comment by paugier** [comment]
I'm going to prepare a first draft for the changelog...

**Comment by hgomersall** [comment]
@paugier Do you want to be added as a maintainer for pyFFTW?

**Comment by paugier** [comment]
> @paugier Do you want to be added as a maintainer for pyFFTW?

Yes, it makes sense and it can indeed be useful that I become maintainer. I really need pyFFTW in a good shape and it's time for me to help the community. Note that I'm a real person :slightly_smiling_face: working at Université Grenoble Alpes, CNRS (http://www.legi.grenoble-inp.fr/people/Pierre.Augier/).

**Comment by hgomersall** [comment]
I'm on vacation for a couple of weeks. If you can make a release with your new rights, go for it, otherwise it might need to wait until I'm back.

**Comment by hgomersall** [comment]
I think we're ready to go

**Comment by paugier** [comment]
If there is no objections, I'll try to push the new tag tomorrow.

**Comment by sahiljhawar** [comment]
v0.14 PyPi CI failed. @paugier can you have a look at it. We ([NMMA](https://github.com/nuclear-multimessenger-astronomy/nmma)) want to ship with Python 3.12 asap which is not possible until 0.14 is available

**Comment by paugier** [comment]
> the most reasonable thing to do now is to download the artifacts and to upload them by hand

Actually, a simpler solution could also be to fix 'heng' account on PyPI (set a verified primary email address) and relaunch the failed CI. But since @hgomersall is on vacation, we anyway need to wait.

**Comment by glaubitz** [comment]
> And I may forget other important things?

You should fix the build with GCC 14 as well, see issue https://github.com/pyFFTW/pyFFTW/issues/378.

**Comment by hgomersall** [comment]
I think the artefacts are handled automatically on a tag - at least that's my current understanding.

**Comment by hgomersall** [comment]
> > And I may forget other important things?
> 
> You should fix the build with GCC 14 as well, see issue #378.

I'd be keen to accept a PR on this prior to release if that can be done. Otherwise a point release.

**Comment by paugier** [comment]
> I think the artefacts are handled automatically on a tag - at least that's my current understanding.

A tag has already been pushed (so we cannot add anything to 0.14.0). There was an issue with uploading the wheels on PyPI. 

```
ERROR    HTTPError: 400 Bad Request from https://upload.pypi.org/legacy/        
         User 'heng' does not have a verified primary email address. Please add 
         a verified primary email before attempting to upload to PyPI. See      
         https://pypi.org/help/#verified-email for more information.
```

@hgomersall could you please try to fix your account on PyPI (verified primary email address) and tell us when it's done?

Then, we will have to rerun the CI associated with the tag 0.14.0.

**Comment by hgomersall** [comment]
> @hgomersall could you please try to fix your account on PyPI (verified primary email address) and tell us when it's done?

Should be fixed now.

**Comment by hgomersall** [comment]
Do we want to fix #378? I'm inclined to crack on and fix it in a point release.

**Comment by sahiljhawar** [comment]
> Do we want to fix #378? I'm inclined to crack on and fix it in a point release.

This sounds good to me, currently all Py3.12 tests are passing, this can be looked in a new PR (although I am not using PyFFTW directly)

**Comment by hgomersall** [comment]
Ach, missed the release notes! It worked though!

**Comment by paugier** [comment]
Yes finally! pyFFTW 0.14.0 wheels are available for users https://pypi.org/project/pyFFTW/0.14.0/, with support for Python 3.12, Numpy 2.0 and Cython 3.0.

Let's hope that everything works as expected. 

For the release notes, we just need to finalize #390 and they should be available on RTD.
